diff --git a/node_modules/react-native-pdf/android/src/main/java/org/wonday/pdf/PdfView.java b/node_modules/react-native-pdf/android/src/main/java/org/wonday/pdf/PdfView.java
index e9a1d16..e8a6d4f 100644
--- a/node_modules/react-native-pdf/android/src/main/java/org/wonday/pdf/PdfView.java
+++ b/node_modules/react-native-pdf/android/src/main/java/org/wonday/pdf/PdfView.java
@@ -310,6 +310,10 @@ public class PdfView extends PDFView implements OnPageChangeListener,OnLoadCompl
             this.setMidZoom((this.maxScale+this.minScale)/2);
             Constants.Pinch.MINIMUM_ZOOM = this.minScale;
             Constants.Pinch.MAXIMUM_ZOOM = this.maxScale;
+            Constants.Cache.CACHE_SIZE = 256;
+            Constants.PRELOAD_OFFSET = 2000;
+            // Render tiles as ARGB_8888 instead of RGB_565 (no color banding on gradients).
+            this.useBestQuality(true);
 
             Configurator configurator;
 
@@ -341,6 +345,7 @@ public class PdfView extends PDFView implements OnPageChangeListener,OnLoadCompl
                 .pageSnap(this.pageSnap)
                 .autoSpacing(this.autoSpacing)
                 .pageFling(this.pageFling)
+                .fitEachPage(true) // <--- THIS FIXES THE MANHWA SCALING
                 .enableSwipe(!this.singlePage && this.scrollEnabled)
                 .enableDoubletap(!this.singlePage && this.enableDoubleTapZoom)
                 .enableAnnotationRendering(this.enableAnnotationRendering)
@@ -354,11 +359,8 @@ public class PdfView extends PDFView implements OnPageChangeListener,OnLoadCompl
                     for (int i=0; i<pageCount; i++) {
                         reversedPages[i] = pageCount-1 - i;
                     }
                     configurator.pages(reversedPages);
-                    if(this.page != 1){
-                        this.page = pageCount;
-                    }
                 } catch (IOException e) {
                     Log.e("error", "error while reading PDF", e);
                 }
             }
